src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/spreadsheet/PriceSheet.kt

```diff
@@ -1,94 +1,181 @@
 package uk.gov.justice.digital.hmpps.pecs.jpc.spreadsheet
 
+import org.apache.pdfbox.util.Hex
 import org.apache.poi.ss.usermodel.BuiltinFormats
 import org.apache.poi.ss.usermodel.CellStyle
 import org.apache.poi.ss.usermodel.FillPatternType
 import org.apache.poi.ss.usermodel.HorizontalAlignment
 import org.apache.poi.ss.usermodel.IndexedColors
 import org.apache.poi.ss.usermodel.Row
 import org.apache.poi.ss.usermodel.Sheet
+import org.apache.poi.ss.util.CellRangeAddress
+import org.apache.poi.xssf.usermodel.XSSFCellStyle
+import org.apache.poi.xssf.usermodel.XSSFColor
 import uk.gov.justice.digital.hmpps.pecs.jpc.move.Journey
 import uk.gov.justice.digital.hmpps.pecs.jpc.move.Move
 import uk.gov.justice.digital.hmpps.pecs.jpc.price.Supplier
 import java.time.LocalDate
 import java.util.concurrent.atomic.AtomicInteger
+import kotlin.math.max
 
-abstract class PriceSheet(val sheet: Sheet, private val header: Header) {
+abstract class PriceSheet(
+  val sheet: Sheet,
+  private val header: Header,
+  private val subheading: String? = null,
```

Comment:
Subheading and dataColumns are new additions.  These are effectively what what is the template spreadsheet.  Now that the whole spreadsheet is entirely generated from code these now live here.